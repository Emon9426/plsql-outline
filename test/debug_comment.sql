-- 简单的注释测试
/* 这是一个多行注释
 * CREATE OR REPLACE PACKAGE BODY fake_package IS
 *   FUNCTION fake_func RETURN NUMBER IS
 *   BEGIN
 *     RETURN 1;
 *   END;
 * END fake_package;
 */

CREATE OR REPLACE FUNCTION real_function RETURN NUMBER IS
BEGIN
    RETURN 42;
END real_function;
/

/* 文件末尾注释
 * CREATE OR REPLACE TRIGGER fake_trigger
 *   BEFORE UPDATE ON fake_table
 * BEGIN
 *   NULL;
 * END;
 */
