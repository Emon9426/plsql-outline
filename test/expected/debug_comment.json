{
  "nodes": [
    {
      "type": "FUNCTION",
      "name": "real_function",
      "declarationLine": 11,
      "beginLine": 12,
      "exceptionLine": null,
      "endLine": 14,
      "level": 1,
      "children": []
    }
  ],
  "metadata": {
    "sourceFile": "debug_comment.sql",
    "totalLines": 5,
    "maxNestingDepth": 1
  }
}